### PR TITLE
registry/storage/driver/s3-aws kms support

### PR DESCRIPTION
**Pull request by matt-duch**
#1518

### CODE REVIEW

**Comment by RichardScothern** [comment]
Thanks for the contribution @matt-duch .  Can you update the documentation for this change? (`docs/configuration.md`)

**Comment by BrianBland** [comment]
Looks like you'll have to rebase this change too, just merged another feature in the S3 driver

**Comment by matt-duch** [comment]
Updated the docs and rebased, all checks passed. Let me know if there's anything else.

**Comment by RichardScothern** [comment]
LGTM

File: docs/storage-drivers/s3.md [hunk 1]
```diff
@@ -89,6 +89,18 @@ An implementation of the `storagedriver.StorageDriver` interface which uses Amaz
   </tr>
   <tr>
     <td>
+      <code>keyid</code>
```
Comment (BrianBland) [inline docs/storage-drivers/s3.md]: Could you also update the parameters list below with a similar entry for `keyid`?

**Comment by matt-duch** [comment]
Updated

**Comment by BrianBland** [comment]
LGTM

**Pull request merged**